app/validators/empty_profile_field_names_validator.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class EmptyProfileFieldNamesValidator < ActiveModel::Validator
+  def validate(account)
+    return if account.fields.empty?
+
+    account.errors.add(:fields, I18n.t('edit_profile.errors.fields_with_values_missing_names')) if fields_with_values_missing_names?(account)
```

Comment:
It is better to use just the key of the message here (`:fields_with_values_missing_names`). Rails will look up the message in `activerecord.en.yml`, for example see `activerecord.errors.models.account.attributes.username.invalid`. You would put your message in `activerecord.errors.models.account.attributes.fields.fields_with_values_missing_names` and build a sentence assuming it starts with the word "Fields", e.g. "contains values with missing names".

By the way, I would double check if the UI uses "names" or "labels", and make sure to use the one already being used for consistency.

Reply:
Thanks, learning a lot about Rails today! I've moved the error message into `activerecord.en.yml` and updated the message to use "labels" as the UI does, but I left the actual validation references with "names" since that's the name of the actual hash key it's checking. I think that lines up with your comment, but please let me know if not.